src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Deposit/components/Deposit.tsx

```diff
@@ -388,6 +568,24 @@ export const Deposit: React.FC<DepositProps> = ({
 
       const quote = maybeQuote.unwrap()
       const { slippage_bps, expected_amount_out: expectedAmountOutThorBaseUnit } = quote
+
+      if (isTokenDeposit) {
+        const daemonUrl = getConfig().REACT_APP_THORCHAIN_NODE_URL
+        const maybeInboundAddressData = await getInboundAddressDataForChain(
+          daemonUrl,
+          feeAsset.assetId,
+        )
+        if (maybeInboundAddressData.isErr())
+          throw new Error(maybeInboundAddressData.unwrapErr().message)
+
+        const inboundAddressData = maybeInboundAddressData.unwrap()
+
+        // This should never happen as THOR *should* return us a router for EVM chains native assets' inbound addresses
+        if (!inboundAddressData.router)
+          throw new Error(`No router address found for feeAsset ${feeAsset.assetId}`)
+        setSaversRouterContractAddress(inboundAddressData.router!)
```

Comment:
theres some repeats of this too, might be cleaner wrapped up in a hook

Reply:
Good call! 69dfa1b80188eb45ed4ca9fc392dbb57b716c5ea
Note how this lives in swapper domain as this relates to THOR as a swapper in a more general sense, not to savers specifically.